src/modules/billing/services/transactions-service.js

```diff
@@ -113,6 +114,32 @@ const mapChargeToTransactions = (chargeLine, batch) => {
   }, []);
 };
 
+/**
+ * Maps agreements array to fields in water.billing_transactions table
+ * @param {Array<Agreement>} agreements
+ * @return {Object}
+ */
+const mapAgreementsToDB = agreements => {
+  const skeleton = {
+    section_127_agreement: false,
+    section_126_factor: 1,
+    section_130_agreement: null
+  };
+
+  return agreements.reduce((acc, agreement) => {
+    if (agreement.code === 'S127') {
+      acc.section_127_agreement = true;
+    }
+    if (agreement.code === 'S126') {
+      acc.section_126_factor = agreement.factor;
+    }
+    if (agreement.code.startsWith('S130')) {
+      acc.section_130_agreement = agreement.code;
+    }
+    return acc;
+  }, skeleton);
+};
+
```

Comment:
I think it would be nice to add functions to the `Agreement` classes for:

 - `isTwoPartTariff()`
 - `isAbatement()`
 - `isCanalAndRiversTrust()`

so we don't have to deal with the codes as much

e.g.

```

const mapAgreementsToDB = agreements => {
  const twoPartTariffAgreement = agreements.find(agreement => agreement.isTwoPartTariff());
  const abatementAgreement = agreements.find(agreement => agreement.isAbatement());
  const canalAndRiversTrustAgreement = agreements.find(agreement => agreement.isCanalAndRiversTrust());

  return {
    section_127_agreement: !!twoPartTariffAgreement,
    section_126_factor: abatementAgreement ? abatementAgreement.factor : 0,
    section_130_agreement: canalAndRiversTrustAgreement ? canalAndRiversTrustAgreement.code : null
  }
};
```

Reply:
Thanks - I have made this change.
I also made another subtle change - the section_126_factor is now stored as null rather than 1 if no section 126 agreement is present.
Although the factor is still sent to the CM as 1, I thought this was clearer.